### PR TITLE
Refactor mibig related code

### PR DESCRIPTION
**Pull request by CunliangGeng**
_Please first review #85. This PR is based on that._

### CODE REVIEW

File: src/nplinker/genomics/mibig/mibig_metadata.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    @property
+    def mibig_accession(self):
+        """Get value of metadata item 'mibig_accession'"""
+        return MibigMetadata.get_mibig_accession(self.metadata)
+
+    @property
+    def biosyn_class(self):
+        """Get value of metadata item 'biosyn_class'"""
+        return MibigMetadata.get_biosyn_class(self.metadata)
```
Comment (hechth) [inline src/nplinker/genomics/mibig/mibig_metadata.py]: Is the biosyn class a known term in the field or something more special? If so, would be nice to have a short explanation or link here.

File: tests/test_mibig_downloader.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    def temppath1(self):
+        temppath = tempfile.mkdtemp()
+        yield temppath
+        shutil.rmtree(temppath)
+
+    @pytest.fixture
+    def temppath2(self):
+        temppath = tempfile.mkdtemp()
+        yield temppath
+        shutil.rmtree(temppath)
```
Comment (hechth) [inline tests/test_mibig_downloader.py]: Actually pytest has a built-in fixture for temporary paths: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

**Review by hechth — APPROVED** [review]

File: src/nplinker/genomics/mibig/mibig_metadata.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/genomics/mibig/mibig_metadata.py]: yes, it's a known term in Mibig metadata, it means "biosynthetic class(es)". I also found it not clear, will add a short explanation for it.

File: tests/test_mibig_downloader.py (hunk 2 above)
Comment (CunliangGeng) [reply tests/test_mibig_downloader.py]: Nice! will take a look to see if it can replace some of my code

**Pull request merged**